pennylane/templates/subroutines/mod_exp.py

```diff
@@ -103,6 +108,10 @@ def __init__(
                 )
         if any(wire in x_wires for wire in output_wires):
             raise ValueError("None of the wires in x_wires should be included in output_wires.")
+
+        if np.gcd(base, mod) != 1:
+            raise ValueError("The operator cannot be built because base has no inverse modulo mod.")
```

Comment:
otherwise the error gets raised later wrt `k`, but if you are using `ModExp`, you haven't used any argument `k`, so I think its cleaner to check here